internal/backends/postgresql/metadata/registry.go

```diff
@@ -65,6 +81,32 @@ func NewRegistry(u string, l *zap.Logger, sp *state.Provider) (*Registry, error)
 		colls: map[string]map[string]*Collection{},
 	}
 
+	// use defer to close registry upon error to avoid forgetting
+	defer func() {
+		if err != nil {
+			r.Close()
+		}
+	}()
+
+	connInfo := conninfo.New()
+	ctx := conninfo.Ctx(context.Background(), connInfo)
+
+	dbPool, err := r.getPool(ctx)
```

Comment:
If PostgreSQL requires authentication and FerretDB's PostgreSQL URL does not contain a username/password, that would fail. But it does not fail now, and this is supported and documented configuration: https://docs.ferretdb.io/security/authentication/ So that's no-go.

`pool` package already stores connection pulls by full URL (that includes authentication). We should do the same there (or move a part of that logic to the pool). Each user / each URL will have its own view of the metadata.

Reply:
> Each user / URL will have its own view of the metadata.

Actually, let's drop that last part – we don't support authorization yet, and we will not support backend-layer authorization going forward. So, let's keep the same metadata for all connected clients, as the Registry's documentation comment says. We still need to delay the initial loading of metadata until we can authenticate – and that means until the first call with conninfo.

Reply:
No-no, let's not change the pool. Let's also store everything in the `colls` field as it is now. What we should do is to call the "load metadata if not loaded yet" method from all other methods that are called from the backend – they will pass conninfo in the context

Reply:
I see, so we don't load `colls` at the creation of registry. Instead we load it upon first call to the registry :+1:

Reply:
Now metadata is loaded upon first call by client :+1: